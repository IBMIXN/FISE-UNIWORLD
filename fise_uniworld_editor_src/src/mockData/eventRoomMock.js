const mockEventRoom = {
  title: "mockTitle",
  eventDate: "2021-03-17T00:00:00.000Z",
  meetingTables: [],
  background: "mockBackground",
  scene: "mockScene",
};

const mockEventRoomUpdate = {
  title: "newMockTitle",
  eventDate: "2021-02-27T12:00:00.000Z",
  meetingTables: [],
  background: "newMockBackground",
  scene: "newMockScene",
};

const mockMeetingTable = {
  title: "mockTitle",
  type: "RoundMeetingTable",
  logoUrl: "https://via.placeholder.com/150",
  zoomUrl: "https://zoom.us/",
  posX: 0,
  posY: 0,
};

export { mockEventRoom, mockEventRoomUpdate, mockMeetingTable };
