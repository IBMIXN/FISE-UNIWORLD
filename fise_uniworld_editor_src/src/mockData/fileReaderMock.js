export default class FileReaderMock {
  onerror = jest.fn();
  onload = jest.fn();
  addEventListener = (_, fn) => fn({ target: { result: {} } });
  readAsDataURL = jest.fn();
}
