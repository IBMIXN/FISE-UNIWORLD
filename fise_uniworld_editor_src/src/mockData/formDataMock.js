export default class FormDataMock {
  append = jest.fn();
}
