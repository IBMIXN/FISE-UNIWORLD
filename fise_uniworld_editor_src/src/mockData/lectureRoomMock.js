const mockLectureRoom = {
  title: "mockTitle",
  module: "mockModule",
  lecturer: "mockLecturer",
  startTime: "2021-03-17T09:00:00.000Z",
  endTime: "2021-03-17T10:00:00.000Z",
  firstSlideUrl: "mockUrl",
  numSlides: 1,
};

const mockLectureRoomUpdate = {
  title: "newMockTitle",
  module: "newMockModule",
  lecturer: "newMockLecturer",
  startTime: "2021-02-27T14:00:00.000Z",
  endTime: "2021-02-27T15:00:00.000Z",
  firstSlideUrl: "newMockUrl",
  numSlides: 3,
};

export { mockLectureRoom, mockLectureRoomUpdate };
