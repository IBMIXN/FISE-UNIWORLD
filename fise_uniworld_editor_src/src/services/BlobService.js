import { axiosConfig } from "../config";

class BlobService {
  url = "/blobs";

  async get(container) {
    return (await axiosConfig.get(`${this.url}/${container}`)).data.blobs;
  }

  async upload(data, container) {
    const config = {
      headers: {
        "content-type": "multipart/form-data",
      },
    };
    return (await axiosConfig.post(`${this.url}/${container}`, data, config)).data.url;
  }

  async delete(fileName, container) {
    const data = { fileName };
    return await axiosConfig.delete(`${this.url}/${container}`, { data });
  }

  async deleteSlides(fileName, numSlides) {
    const data = { fileName, numSlides };
    return await axiosConfig.delete(`${this.url}/slides`, { data });
  }
}

export default new BlobService();
