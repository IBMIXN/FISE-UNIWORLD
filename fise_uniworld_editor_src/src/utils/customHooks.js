import React from "react";

const useReducerState = (defaultState) => {
  const reducer = (state, newState) => ({ ...state, ...newState });
  return React.useReducer(reducer, defaultState);
};

export { useReducerState };
