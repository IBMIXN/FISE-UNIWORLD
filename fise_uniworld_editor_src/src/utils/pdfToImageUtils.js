const PDFJS = require("pdfjs-dist/webpack");

const getCanvasBlobAsync = (canvas) => {
  return new Promise((resolve, reject) => {
    canvas.toBlob((blob) => {
      resolve(blob);
    });
  });
};

const readFileData = (file) => {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    // reader.onload = (e) => {
    //   resolve(e.target.result);
    // };
    reader.addEventListener("load", (e) => {
      resolve(e.target.result);
    });
    reader.onerror = (err) => {
      reject(err);
    };
    reader.readAsDataURL(file);
  });
};

const convertPdfToImages = async (file) => {
  const data = await readFileData(file);
  const pdf = await PDFJS.getDocument(data).promise;
  const formData = new FormData();
  const canvas = document.createElement("canvas");
  const numSlides = pdf.numPages;
  for (let i = 0; i < pdf.numPages; i++) {
    const page = await pdf.getPage(i + 1);
    const viewport = page.getViewport({ scale: 1 });
    const context = canvas.getContext("2d");
    canvas.height = viewport.height;
    canvas.width = viewport.width;
    await page.render({ canvasContext: context, viewport: viewport }).promise;
    const fileNameWithoutExt = file.name.substring(0, file.name.lastIndexOf("."));
    const newFileName = fileNameWithoutExt + "_" + i + ".png";
    const slideImageBlob = await getCanvasBlobAsync(canvas);
    formData.append(`images[${i}]`, slideImageBlob, newFileName);
  }
  canvas.remove();

  return {
    numSlides,
    formData,
  };
};

const convertPreviewImage = async (file) => {
  const data = await readFileData(file);
  const pdf = await PDFJS.getDocument(data).promise;
  const canvas = document.createElement("canvas");
  const page = await pdf.getPage(1);
  const viewport = page.getViewport({ scale: 1 });
  const context = canvas.getContext("2d");
  canvas.height = viewport.height;
  canvas.width = viewport.width;
  await page.render({ canvasContext: context, viewport: viewport }).promise;
  canvas.remove();
  return canvas.toDataURL();
};
export { convertPdfToImages, convertPreviewImage };
