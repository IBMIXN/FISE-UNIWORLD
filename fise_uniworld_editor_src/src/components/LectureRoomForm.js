import PropTypes from "prop-types";
import { UpdateLectureRoomPropType } from "../propTypes/LectureRoom";
import React from "react";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import {
  Grid,
  Typography,
  Card,
  CardActions,
  CardContent,
  Button,
  TextField,
  CircularProgress,
} from "@material-ui/core";
import MomentUtils from "@date-io/moment";
import { MuiPickersUtilsProvider, KeyboardDateTimePicker } from "@material-ui/pickers";
import AddIcon from "@material-ui/icons/Add";
import FileUpload from "../components/FileUpload";
import { useReducerState } from "../utils/customHooks";
import BlobService from "../services/BlobService";
import { convertPdfToImages, convertPreviewImage } from "../utils/pdfToImageUtils";

const useStyles = makeStyles((theme) => ({
  lectureInfoPanel: {
    marginTop: theme.spacing(1),
    padding: theme.spacing(2),
  },
  divider: {
    padding: theme.spacing(1),
  },
  centerTableGrid: {
    display: "flex",
    justifyContent: "center",
  },
  createButton: {
    "&:hover": {
      backgroundColor: theme.palette.success.dark,
    },
    backgroundColor: theme.palette.success.main,
    color: "#fff",
  },
  addIcon: {
    color: "#fff",
    marginRight: theme.spacing(2),
  },
  progressLoader: {
    color: "#fff",
    marginRight: theme.spacing(2),
  },
  padTop: {
    paddingTop: theme.spacing(1),
  },
  padRight: {
    paddingRight: theme.spacing(1),
  },
  flexDisplay: {
    display: "flex",
  },
}));

const LectureRoomForm = ({ onSubmit, submitText, updateLectureRoom }) => {
  const classes = useStyles();
  const [isLoading, setLoading] = React.useState(false);
  const lectureRoomInitialState = {
    title: "",
    module: "",
    lecturer: "",
    startTime: moment().toISOString(),
    endTime: moment().toISOString(),
    firstSlideUrl: "",
    numSlides: 1,
  };
  const [lectureRoom, setLectureRoom] = useReducerState(lectureRoomInitialState);
  const [file, setFile] = React.useState(null);
  const [previewImage, setPreviewImage] = React.useState(null);

  React.useEffect(() => {
    if (updateLectureRoom) {
      setLectureRoom(updateLectureRoom);
    }
  }, [updateLectureRoom, setLectureRoom]);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setLectureRoom({
      [name]: value,
    });
  };

  const handleUploadLecturePdf = async (e) => {
    e.preventDefault();
    if (e.target.files[0] && e.target.files[0].name.match(/\.(pdf)$/)) {
      setFile(e.target.files[0]);
      setPreviewImage(await convertPreviewImage(e.target.files[0]));
    } else {
      alert("Please select a valid PDF document.");
    }
  };

  const handleDateChange = (val, start) => {
    if (start) {
      setLectureRoom({
        startTime: moment(val).toISOString(),
      });
    } else {
      setLectureRoom({
        endTime: moment(val).toISOString(),
      });
    }
  };

  const handleClickSubmit = async () => {
    setLoading(true);
    let data = { ...lectureRoom };
    if (file) {
      const { numSlides, formData } = await convertPdfToImages(file);
      const slide = await BlobService.upload(formData, "slides");
      data = { ...data, numSlides, firstSlideUrl: slide };
    }
    await onSubmit(data);
    setLectureRoom(lectureRoomInitialState);
    setLoading(false);
  };

  return (
    <Grid container spacing={2}>
      {" "}
      <Grid item md={12}>
        <Card className={classes.lectureInfoPanel}>
          <React.Fragment>
            <CardContent>
              <Grid container spacing={3}>
                <Grid item md={6} xs={12}>
                  <TextField
                    fullWidth
                    disabled={isLoading}
                    id="title"
                    name="title"
                    label="Title"
                    placeholder="Enter the lecture title"
                    value={lectureRoom.title}
                    onChange={handleChange}
                    InputLabelProps={{
                      shrink: true,
                    }}
                  />
                  <div className={classes.divider} />
                  <TextField
                    fullWidth
                    disabled={isLoading}
                    id="module"
                    name="module"
                    label="Module"
                    placeholder="Enter the module"
                    value={lectureRoom.module}
                    onChange={handleChange}
                    InputLabelProps={{
                      shrink: true,
                    }}
                  />
                  <div className={classes.divider} />
                  <TextField
                    fullWidth
                    disabled={isLoading}
                    id="lecturer"
                    name="lecturer"
                    label="Lecturer"
                    placeholder="Enter lecturer name"
                    value={lectureRoom.lecturer}
                    onChange={handleChange}
                    InputLabelProps={{
                      shrink: true,
                    }}
                  />
                  <MuiPickersUtilsProvider utils={MomentUtils}>
                    <KeyboardDateTimePicker
                      id="startTime"
                      fullWidth
                      disabled={isLoading}
                      disablePast
                      showTodayButton
                      format="h:mm a - YYYY/MM/DD"
                      margin="normal"
                      label="Start Time"
                      value={lectureRoom.startTime}
                      onChange={(val) => {
                        handleDateChange(val, true);
                      }}
                      KeyboardButtonProps={{
                        "aria-label": "change start time",
                      }}
                    />
                  </MuiPickersUtilsProvider>
                  <MuiPickersUtilsProvider utils={MomentUtils}>
                    <KeyboardDateTimePicker
                      id="endTime"
                      fullWidth
                      disabled={isLoading}
                      disablePast
                      showTodayButton
                      format="h:mm a - YYYY/MM/DD"
                      margin="normal"
                      label="End Time"
                      value={lectureRoom.endTime}
                      onChange={(val) => {
                        handleDateChange(val, false);
                      }}
                      KeyboardButtonProps={{
                        "aria-label": "change end time",
                      }}
                    />
                  </MuiPickersUtilsProvider>
                  <div className={classes.divider} />
                  <div className={classes.flexDisplay}>
                    <TextField
                      fullWidth
                      className={classes.padRight}
                      disabled={isLoading}
                      name="firstSlideurl"
                      label="Lecture Slides"
                      placeholder="Upload lecture slides in a PDF format"
                      value={file ? file.name : lectureRoom.firstSlideUrl || ""}
                      onChange={() => {}}
                      InputLabelProps={{
                        shrink: true,
                      }}
                      InputProps={{
                        readOnly: true,
                      }}
                    />
                    <FileUpload
                      id="lectureSlideUpload"
                      name="lectureSlideUpload"
                      onChange={handleUploadLecturePdf}
                      width="120px"
                      accept="application/pdf"
                    >
                      Upload
                    </FileUpload>
                  </div>
                </Grid>
                <Grid item md={6} xs={12}>
                  <Typography variant="subtitle2">Slide Preview</Typography>
                  <img
                    width="100%"
                    alt="Slide Preview"
                    src={
                      previewImage ||
                      lectureRoom.firstSlideUrl ||
                      "https://via.placeholder.com/700x500.png?text=Placeholder+Slide"
                    }
                  />
                </Grid>
              </Grid>
            </CardContent>
            <CardActions>
              <Button
                id="submitButton"
                className={classes.createButton}
                color="primary"
                variant="outlined"
                size="large"
                onClick={handleClickSubmit}
                disabled={isLoading}
              >
                {isLoading ? (
                  <CircularProgress className={classes.progressLoader} size={20} />
                ) : (
                  <AddIcon className={classes.addIcon} />
                )}
                {submitText}
              </Button>
            </CardActions>
          </React.Fragment>
        </Card>
      </Grid>
    </Grid>
  );
};

LectureRoomForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitText: PropTypes.string.isRequired,
  updateLectureRoom: UpdateLectureRoomPropType,
};

LectureRoomForm.defaultProps = {
  updateLectureRoom: undefined,
};

export default LectureRoomForm;
