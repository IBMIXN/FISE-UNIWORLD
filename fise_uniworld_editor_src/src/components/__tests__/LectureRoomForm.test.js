import React from "react";
import { shallow } from "enzyme";
import LectureRoomForm from "../LectureRoomForm";
import { mockLectureRoomUpdate } from "../../mockData/lectureRoomMock";
import BlobService from "../../services/BlobService";
import { useReducerState } from "../../utils/customHooks";
import { convertPdfToImages, convertPreviewImage } from "../../utils/pdfToImageUtils";
jest.mock("../../services/BlobService");
jest.mock("../../utils/customHooks", () => {
  return {
    useReducerState: jest.fn(),
  };
});
jest.mock("../../utils/pdfToImageUtils", () => {
  return {
    convertPdfToImages: jest.fn(),
    convertPreviewImage: jest.fn(),
  };
});

describe("LectureRoomForm test", () => {
  const lectureRoom = {
    title: "",
    module: "",
    lecturer: "",
    startTime: "",
    endTime: "",
    firstSlideUrl: "",
    numSlides: 1,
  };
  const setLectureRoom = jest.fn();
  const flushPromises = () => new Promise(setImmediate);

  beforeAll(() => {
    Date.now = jest.fn(() => new Date("2021-03-01T12:00:00.000Z"));
  });

  let mockAlert;
  beforeEach(() => {
    useReducerState.mockImplementation(() => [lectureRoom, setLectureRoom]);
    jest.spyOn(console, "log").mockImplementation(() => {});
    mockAlert = jest.spyOn(window, "alert").mockImplementation(() => {});
  });

  it("should match snapshot", () => {
    const component = shallow(
      <LectureRoomForm
        onSubmit={() => {}}
        submitText="mockText"
        updateLectureRoom={mockLectureRoomUpdate}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should correctly render preview image for firstSlideUrl", () => {
    useReducerState.mockImplementation(() => [
      { ...lectureRoom, firstSlideUrl: "mockSlideUrl" },
      setLectureRoom,
    ]);
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    expect(component).toMatchSnapshot();
  });

  it("should set lectureRoom if updateLectureRoom prop exists", () => {
    shallow(
      <LectureRoomForm
        onSubmit={() => {}}
        submitText="mockText"
        updateLectureRoom={mockLectureRoomUpdate}
      />
    );
    expect(setLectureRoom).toBeCalledWith(mockLectureRoomUpdate);
  });

  it("should not set lectureRoom if updateLectureRoom prop is undefined", () => {
    shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    expect(setLectureRoom).not.toBeCalled();
  });

  it("should correctly handle title text change event", () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { name: "title", value: "mockTitle" } };
    component.find("#title").first().simulate("change", mockEvent);
    expect(setLectureRoom).toBeCalledWith({ title: mockEvent.target.value });
  });

  it("should correctly handle module text change event", () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { name: "module", value: "mockModule" } };
    component.find("#module").first().simulate("change", mockEvent);
    expect(setLectureRoom).toBeCalledWith({ module: mockEvent.target.value });
  });

  it("should correctly handle lecturer text change event", () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { name: "lecturer", value: "mockLecturer" } };
    component.find("#lecturer").first().simulate("change", mockEvent);
    expect(setLectureRoom).toBeCalledWith({ lecturer: mockEvent.target.value });
  });

  it("should correctly handle start time change event", () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockStartTime = new Date().toISOString();
    component.find("#startTime").first().simulate("change", mockStartTime);
    expect(setLectureRoom).toBeCalledWith({ startTime: mockStartTime });
  });

  it("should correctly handle end time change event", () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEndTime = new Date().toISOString();
    component.find("#endTime").first().simulate("change", mockEndTime);
    expect(setLectureRoom).toBeCalledWith({ endTime: mockEndTime });
  });

  it("should correctly handle upload lecture slide", async () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { files: [{ name: "mockSlide.pdf" }] }, preventDefault: () => {} };
    component.find("#lectureSlideUpload").first().simulate("change", mockEvent);
    await flushPromises();
    expect(convertPreviewImage).toBeCalledWith(mockEvent.target.files[0]);
  });

  it("should alert user if invalid (non-pdf) file is uploaded", async () => {
    const component = shallow(<LectureRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = {
      target: { files: [{ name: "invalidFile.jpg" }] },
      preventDefault: () => {},
    };
    component.find("#lectureSlideUpload").first().simulate("change", mockEvent);
    await flushPromises();
    expect(convertPreviewImage).not.toBeCalled();
    expect(mockAlert).toBeCalledWith("Please select a valid PDF document.");
  });

  it("should correctly handle submit form", async () => {
    convertPdfToImages.mockImplementationOnce(() =>
      Promise.resolve({ numSlides: 0, formData: null })
    );
    BlobService.upload.mockImplementationOnce(() => Promise.resolve("mockSlide"));
    const mockSubmit = jest.fn();
    const component = shallow(
      <LectureRoomForm onSubmit={mockSubmit} submitText="mockText" updateLectureRoom={{}} />
    );
    const mockEvent = { target: { files: [{ name: "mockSlide.pdf" }] }, preventDefault: () => {} };
    component.find("#lectureSlideUpload").first().simulate("change", mockEvent);
    await flushPromises();
    component.find("#submitButton").first().simulate("click");
    await flushPromises();
    expect(convertPdfToImages).toBeCalledWith(mockEvent.target.files[0]);
    expect(BlobService.upload.mock.calls[0][1]).toBe("slides");
    expect(mockSubmit).toBeCalledWith({ ...lectureRoom, numSlides: 0, firstSlideUrl: "mockSlide" });
    expect(setLectureRoom).toBeCalled();
  });

  it("should correctly handle submit form when no files are uploaded", async () => {
    const mockSubmit = jest.fn();
    const component = shallow(
      <LectureRoomForm onSubmit={mockSubmit} submitText="mockText" updateLectureRoom={{}} />
    );
    component.find("#submitButton").first().simulate("click");
    await flushPromises();
    expect(convertPdfToImages).not.toBeCalled();
    expect(BlobService.upload).not.toBeCalled();
    expect(mockSubmit).toBeCalled();
    expect(setLectureRoom).toBeCalled();
  });
});
