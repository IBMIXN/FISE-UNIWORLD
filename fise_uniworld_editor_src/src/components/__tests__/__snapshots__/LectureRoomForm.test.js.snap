// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LectureRoomForm test should correctly render preview image for firstSlideUrl 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
   
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-lectureInfoPanel-1"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={6}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="title"
              label="Title"
              name="title"
              onChange={[Function]}
              placeholder="Enter the lecture title"
              value=""
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="module"
              label="Module"
              name="module"
              onChange={[Function]}
              placeholder="Enter the module"
              value=""
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="lecturer"
              label="Lecturer"
              name="lecturer"
              onChange={[Function]}
              placeholder="Enter lecturer name"
              value=""
            />
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change start time",
                  }
                }
                allowKeyboardControl={true}
                ampm={true}
                disablePast={true}
                disabled={false}
                format="h:mm a - YYYY/MM/DD"
                fullWidth={true}
                id="startTime"
                invalidDateMessage="Invalid Date Format"
                label="Start Time"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                orientation="portrait"
                showTabs={true}
                showTodayButton={true}
                value=""
                views={
                  Array [
                    "year",
                    "date",
                    "hours",
                    "minutes",
                  ]
                }
                wider={true}
              />
            </MuiPickersUtilsProvider>
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change end time",
                  }
                }
                allowKeyboardControl={true}
                ampm={true}
                disablePast={true}
                disabled={false}
                format="h:mm a - YYYY/MM/DD"
                fullWidth={true}
                id="endTime"
                invalidDateMessage="Invalid Date Format"
                label="End Time"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                orientation="portrait"
                showTabs={true}
                showTodayButton={true}
                value=""
                views={
                  Array [
                    "year",
                    "date",
                    "hours",
                    "minutes",
                  ]
                }
                wider={true}
              />
            </MuiPickersUtilsProvider>
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-flexDisplay-9"
            >
              <WithStyles(ForwardRef(TextField))
                InputLabelProps={
                  Object {
                    "shrink": true,
                  }
                }
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                className="makeStyles-padRight-8"
                disabled={false}
                fullWidth={true}
                label="Lecture Slides"
                name="firstSlideurl"
                onChange={[Function]}
                placeholder="Upload lecture slides in a PDF format"
                value="mockSlideUrl"
              />
              <FileUpload
                accept="application/pdf"
                id="lectureSlideUpload"
                name="lectureSlideUpload"
                onChange={[Function]}
                width="120px"
              >
                Upload
              </FileUpload>
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={6}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="subtitle2"
            >
              Slide Preview
            </WithStyles(ForwardRef(Typography))>
            <img
              alt="Slide Preview"
              src="mockSlideUrl"
              width="100%"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-4"
          color="primary"
          disabled={false}
          id="submitButton"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          <Memo([object Object])
            className="makeStyles-addIcon-5"
          />
          mockText
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`LectureRoomForm test should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
   
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-lectureInfoPanel-1"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={6}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="title"
              label="Title"
              name="title"
              onChange={[Function]}
              placeholder="Enter the lecture title"
              value=""
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="module"
              label="Module"
              name="module"
              onChange={[Function]}
              placeholder="Enter the module"
              value=""
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="lecturer"
              label="Lecturer"
              name="lecturer"
              onChange={[Function]}
              placeholder="Enter lecturer name"
              value=""
            />
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change start time",
                  }
                }
                allowKeyboardControl={true}
                ampm={true}
                disablePast={true}
                disabled={false}
                format="h:mm a - YYYY/MM/DD"
                fullWidth={true}
                id="startTime"
                invalidDateMessage="Invalid Date Format"
                label="Start Time"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                orientation="portrait"
                showTabs={true}
                showTodayButton={true}
                value=""
                views={
                  Array [
                    "year",
                    "date",
                    "hours",
                    "minutes",
                  ]
                }
                wider={true}
              />
            </MuiPickersUtilsProvider>
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change end time",
                  }
                }
                allowKeyboardControl={true}
                ampm={true}
                disablePast={true}
                disabled={false}
                format="h:mm a - YYYY/MM/DD"
                fullWidth={true}
                id="endTime"
                invalidDateMessage="Invalid Date Format"
                label="End Time"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                orientation="portrait"
                showTabs={true}
                showTodayButton={true}
                value=""
                views={
                  Array [
                    "year",
                    "date",
                    "hours",
                    "minutes",
                  ]
                }
                wider={true}
              />
            </MuiPickersUtilsProvider>
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-flexDisplay-9"
            >
              <WithStyles(ForwardRef(TextField))
                InputLabelProps={
                  Object {
                    "shrink": true,
                  }
                }
                InputProps={
                  Object {
                    "readOnly": true,
                  }
                }
                className="makeStyles-padRight-8"
                disabled={false}
                fullWidth={true}
                label="Lecture Slides"
                name="firstSlideurl"
                onChange={[Function]}
                placeholder="Upload lecture slides in a PDF format"
                value=""
              />
              <FileUpload
                accept="application/pdf"
                id="lectureSlideUpload"
                name="lectureSlideUpload"
                onChange={[Function]}
                width="120px"
              >
                Upload
              </FileUpload>
            </div>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={6}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="subtitle2"
            >
              Slide Preview
            </WithStyles(ForwardRef(Typography))>
            <img
              alt="Slide Preview"
              src="https://via.placeholder.com/700x500.png?text=Placeholder+Slide"
              width="100%"
            />
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-4"
          color="primary"
          disabled={false}
          id="submitButton"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          <Memo([object Object])
            className="makeStyles-addIcon-5"
          />
          mockText
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
